source/slang/slang.natvis

```diff
@@ -15,12 +15,15 @@
     <DisplayString Condition="decl != 0">{(*(*(Slang::DeclRefBase*)this).decl).nameAndLoc}</DisplayString>
     <Expand>
       <ExpandedItem>decl ? ($T1*)(decl) : ($T1*)0</ExpandedItem>
-      <Item Name="[Substitutions]:">"========================="</Item>
-      <LinkedListItems>
-        <HeadPointer>substitutions.substitutions.pointer</HeadPointer>
-        <NextPointer>outer.pointer</NextPointer>
-        <ValueNode>this</ValueNode>
-      </LinkedListItems>
+      <Synthetic Name="[Substitutions]">
```

Comment:
Nice to have better natvis.

Guess I worry about amount of 'stuff' needed to be kept in sync. This is to work around the AST derived types not having a virtual function, such that vs knows the type?

Reply:
Right. It will also save one expansion click since it will just directly display the derived type.